common/filter-tags/outputs.tf

```
@@ -0,0 +1,89 @@
+output "filter_custom" {
+  description = "The full filtering pattern to use in detectors"
+  value = (
+    var.filter_use_defaults == true ?
+    var.filter_defaults
+    :
+    var.filter_custom_includes == "" ?
+    join(
+      " and ",
+      formatlist(
+        "(not filter('%s'))",
+        split(
+          ";",
+          replace(
+            replace(
+              var.filter_custom_excludes,
+              ":",
+              "', '"
+            ),
+            " ",
+            ""
+          )
+        )
+      )
+    )
+    :
+    var.filter_custom_excludes == "" ?
+    join(
+      " and ",
+      formatlist(
+        "(not filter('%s'))",
+        split(
+          ";",
+          replace(
+            replace(
+              var.filter_custom_includes,
+              ":",
+              "', '"
+            ),
+            " ",
+            ""
+          )
+        )
+      )
+    )
+    :
+    join(
+      " and ",
+      list(
+        join(
+          " and ",
+          formatlist(
+            "filter('%s')",
+            split(
+              ";",
+              replace(
+                replace(
+                  var.filter_custom_includes,
+                  ":",
+                  "', '"
+                ),
+                " ",
+                ""
+              )
+            )
+          )
+        ),
+        join(
+          " and ",
+          formatlist(
+            "(not filter('%s'))",
+            split(
+              ";",
+              replace(
+                replace(
+                  var.filter_custom_excludes,
+                  ":",
+                  "', '"
+                ),
+                " ",
+                ""
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+}
```

As discussed, simpler variant (with passed as lists):
```suggestion
locals {
  filter_custom_includes = join(
    " and ",
    formatlist(
      "(filter('%s'))",
      [for tag in var.filter_custom_includes : replace(replace(tag, ":", "', '"), " ", "")]
    )
  )

  filter_custom_excludes = join(
    " and ",
    formatlist(
      "(not filter('%s'))",
      [for tag in var.filter_custom_excludes : replace(replace(tag, ":", "', '"), " ", "")]
    )
  )
}

output "filter_custom" {
  description = "The full filtering pattern to use in detectors"
  value = coalesce(
    join(
      " and ",
      compact(
        [
          local.filter_custom_includes,
          local.filter_custom_excludes,
        ]
      )
    ),
    var.filter_defaults
  )
}
```

---

It is awesome ! I tested it works fine, thank you